### PR TITLE
Feat create device

### LINKED ISSUE #8: Use case for getting a device
Devices are those who can trigger the use cases related with reminders, those could be an NFC tag, bluetooth device, etc, etc, etc.

This use case should:
1) Given the deviceIdentifier, search on the DeviceRepository.
2) If the device exists, return the device, if not, return null.

### PR DESCRIPTION
**Pull request by oriolagobat**
This closes #8

### CODE REVIEW

File: src/main/kotlin/device/application/GetDevice.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+class GetDevice(
+    private val repository: DeviceRepository,
+) {
+    /***
+     * Searches a Device with the given identifier and returns it if found.
+     * @see Device
+     * @return null or a device instance if it exists
+     */
+    fun get(identifier: DeviceIdentifier) = repository.get(identifier)
```
Comment (FerranAD) [inline src/main/kotlin/device/application/GetDevice.kt:15]: This should receive a primitive or a DevicePrimitives object:
```kotlin
    fun get(primitives: DevicePrimitives) = repository.get(Identifier(primitives.identifier))
```
or
```kotlin
    fun get(identifier: String) = repository.get(Identifier(identifier))
```

**Review by FerranAD — COMMENTED** [review]
Usage of primitives instead of domain objects in use case.

File: src/main/kotlin/device/application/GetDevice.kt (hunk 1 above)
Comment (FerranAD) [inline src/main/kotlin/device/application/GetDevice.kt:15]: Like I said on #39, there's no need to use a specific identifier class for Domain classes, Identifier superclass can be used instead.

**Review by FerranAD — CHANGES_REQUESTED** [review]
Usage of Identifier superclass.

File: src/main/kotlin/device/application/GetDevice.kt (hunk 1 above)
Comment (oriolagobat) [reply src/main/kotlin/device/application/GetDevice.kt:15]: Get function shouldn't receive a primitives object, but a Identifier, as that's the only thing we really want.
That would be a good practice to apply on the create and update use case.
Is that correct?

Comment (oriolagobat) [reply src/main/kotlin/device/application/GetDevice.kt:15]: I've refactored the create use case as we said.

**Review by Pablito2020 — APPROVED** [review]

**Review by FerranAD — APPROVED** [review]

**Pull request merged**